### PR TITLE
Implement base64 encode

### PR DESCRIPTION
**Pull request by lizan**
#301

### CODE REVIEW

**Comment by mattklein123** [comment]
@lizan can you please sign the CLA and also have someone else from Google review this.

@RomanDzhabarov can you also take a look.

File: source/common/common/base64.cc [hunk 1]
```diff
@@ -1,35 +1,59 @@
...
-    uint64_t to_write = std::min(length, slice.len_);
-    length -= to_write;
-    BIO_write(bio, slice.mem_, to_write);
+    const uint8_t* slice_mem = static_cast<const uint8_t*>(slice.mem_);
+
+    for (uint64_t i = 0; i < slice.len_ && j < length; ++i, ++j) {
+      const uint8_t c = slice_mem[i];
+      switch (j % 3) {
+      case 0:
+        ret.push_back(CHAR_TABLE[c >> 2]);
```
Comment (qiwzhang) [inline source/common/common/base64.cc]: could "c >> 2" be bigger than sizeof(CHAR_TABLE)?

Comment (qiwzhang) [inline source/common/common/base64.cc]: Never mind. Please discard my previous comment

**Comment by lizan** [comment]
I'm working on CLA, it might take a while.

**Comment by RomanDzhabarov** [comment]
@mattklein123 @lizan I'll check it out today

**Comment by lizan** [comment]
CLA signed! Sorry for the delay.

**Pull request merged**